jazz_assets/components/crud/create.js

```diff
@@ -0,0 +1,66 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+
+/**
+    Create a Asset-Catalog entry in dynamodb table
+    @module: create.js
+    @description: CRUD functions for asset catalog
+    @author:
+    @version: 1.0
+**/
+
+const utils = require("../utils.js")(); //Import the utils module.
+const Uuid = require("uuid/v4");
+const moment = require('moment');
+
+module.exports = (assets_data, onComplete) => {
+    var docClient = utils.initDocClient();
+
+    var assets_id = Uuid();
+    var timestamp = moment().utc().format('YYYY-MM-DDTHH:mm:ss:SSS');
+    var params = {
+        Item: {
+            "id": assets_id,
+            "timestamp": timestamp
+			},
+        ReturnConsumedCapacity: "TOTAL",
+        TableName: global.ASSETS_TABLE
+    };
+
+    Object.keys(assets_data).map((key) => {
```

Comment:
although we have been mutating the array in the map but the right way to use the map is to return the value which create a new array. That is the proper usage of map without side effect. - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
Noted.